src/Templates.hs

```diff
@@ -0,0 +1,61 @@
+module Templates where
+
+import qualified Data.HashMap.Strict as HM
+import qualified Data.HashMap.Strict as HashMap
+import Foundation
+import System.IO.Error (tryIOError)
+import Templates.Types (ModuleName (..), TemplateAssigns (..),
+                        TemplateName (..))
+import Text.Ginger (GVal, Source, SourceName, ToGVal (..), makeContextHtml,
+                    parseGingerFile, runGinger)
+import Text.Ginger.Html (htmlSource)
+import Web.Helpers (debug)
+import Web.Sessions (UserAssigns (UserAssigns), getAllUserAssigns, popAssign)
+import Web.Types (MatchmakerError)
+import Yesod.Core
+
+render :: ModuleName -> TemplateName -> TemplateAssigns -> Handler Html
+render (ModuleName moduleName) (TemplateName templateName) assigns = do
+  let templatePath = "./src/Templates/" <> moduleName <> "/" <> templateName <> ".html"
+  mUserAssigns <- getAllUserAssigns
+  let (TemplateAssigns hm) = mkAssigns assigns mUserAssigns
+  debug ("Assigns: " <> show hm)
+  let contextLookup = flip scopeLookup hm
+  let context = makeContextHtml contextLookup
+  eTemplate <- liftIO $ parseGingerFile resolver (toString templatePath)
+  case eTemplate of
+    Left err -> pure $ show err
+    Right template -> do
+      popAssign "flash_alert_info"
+      popAssign "flash_alert_error"
+      pure . preEscapedToMarkup . htmlSource $ runGinger context template
```

Comment:
We need to get to a `MarkupM ()` type here so that Yesod uses the right content type, but ginger pre escapes characters via `htmlSource`, which is why we need to use `preEscapedToMarkup`.